lib/blacklight_advanced_search/controller.rb

```diff
@@ -14,44 +14,35 @@ module BlacklightAdvancedSearch::Controller
     self.blacklight_config.advanced_search[:form_solr_parameters] ||= {}    
     
     
-    # Parse app URL params used for adv searches 
-    self.solr_search_params_logic += [:add_advanced_search_to_solr]
-    
+    if self.respond_to? :search_params_logic
+      # Parse app URL params used for adv searches 
+      self.search_params_logic += [:add_advanced_search_to_solr]
+    end
+
+    if self.blacklight_config.search_builder_class
+    self.blacklight_config.search_builder_class.send(:include,  
```

Comment:
Ah, this was your goal in https://github.com/projectblacklight/blacklight/pull/1128? I'm not sure modules make it any better. Currently, couldn't we just generate:

```
class SearchBuilder < Blacklight::Solr::SearchBuilder
  include BlacklightAdvancedSearch::Solr::SearchBuilderStuff # or whatever
end
```